packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -140,6 +140,8 @@ class NavigationRail extends StatefulWidget {
     this.selectedIconTheme,
     this.minWidth,
     this.minExtendedWidth,
+    this.curve = Curves.fastOutSlowIn,
```

Comment:
The curve and duration defaults should be null, like the other constructor parameters. The defaults should be computed based on (new) NavigationRailThemeData properties. Please follow the pattern for defaults established in the build method.

Reply:
I want to pick this up. 

This curve is used to initialize the animation controller in the  `initState` method. Yet, the default values depend on the `BuildContext`. Is there a recommended way to proceed?